### PR TITLE
Xena: Rocky 8 AIO scenario 

### PR DESCRIPTION
**Pull request by jovial**
I'll rebase this when this when the test is passing. It builds on a few other open PRs.

### CODE REVIEW

File: etc/kayobe/overcloud-dib.yml [hunk 1]
```diff
@@ -71,7 +71,8 @@ overcloud_dib_host_packages_extra:
 overcloud_dib_git_elements_extra:
   - repo: "https://github.com/stackhpc/stackhpc-image-elements"
     local: "{{ source_checkout_path }}/stackhpc-image-elememts"
-    version: "v1.4.0"
+    # FIXME: merge and tag a new release
```
Comment (jovial) [inline etc/kayobe/overcloud-dib.yml:74]: Not technically need for AIO, so could this split this out. I was just testing with the image produced by DIB. Should we switch to using the LVM based DIB images at some point?

File: terraform/aio/vm.tf [hunk 2]
```diff
@@ -59,35 +49,9 @@ resource "openstack_compute_instance_v2" "kayobe-aio" {
   block_device {
     uuid                  = data.openstack_images_image_v2.image.id
     source_type           = "image"
-    volume_size           = 100
+    volume_size           = 200
```
Comment (jovial) [inline terraform/aio/vm.tf:61]: This was for the stackhpc LVM layout with the DIB image

**Comment by jovial** [comment]
Is it best to land this change on wallaby? Seems easier to merge up than down? I assumed I should target the latest release, but maybe this was a mistake?

**Comment by markgoddard** [comment]
> Is it best to land this change on wallaby? Seems easier to merge up than down? I assumed I should target the latest release, but maybe this was a mistake?

It is possible to merge up, but not down. You can cherry pick down if necessary though. It might be safer to start with Xena since it's not in use yet.

Will we have all the necessary dependencies in Wallaby?

File: etc/kayobe/ansible/templates/stackhpc-secrets.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+# SASL password when using libvirt on the host
+secrets_libvirt_sasl_password: "{{ secrets_libvirt_sasl_password | default(lookup('password', '/dev/null'), true) }}"
```
Comment (markgoddard) [inline etc/kayobe/ansible/templates/stackhpc-secrets.yml.j2]: Could you not use the password lookup plugin, and write the result to a file?

File: etc/kayobe/compute.yml [hunk 4]
```diff
@@ -179,6 +179,15 @@
 # compute_qemu_conf_extra.
 #compute_qemu_conf:
 
+# Whether to enable libvirt SASL authentication. Default is true.
```
Comment (markgoddard) [inline etc/kayobe/compute.yml:182]: Do we need a config sync somewhere?

File: etc/kayobe/compute.yml [hunk 5]
```diff
@@ -179,6 +179,15 @@
...
+#compute_libvirt_enable_sasl:
+
+# libvirt SASL password. Default is unset.
+# NOTE: You can use:
+#   kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/generate-stackhpc-secrets.yml
+# to generate secrets_libvirt_sasl_password.
+compute_libvirt_sasl_password: "{{ secrets_libvirt_sasl_password }}"
```
Comment (markgoddard) [inline etc/kayobe/compute.yml]: Probably needs to be in the ci-aio environment?

File: etc/kayobe/kolla.yml [hunk 6]
```diff
@@ -506,6 +506,7 @@ kolla_build_args:
 #kolla_enable_kibana:
 #kolla_enable_kuryr:
 #kolla_enable_loadbalancer:
+kolla_enable_nova_libvirt_container: "{{ not os_distribution in ['rocky'] }}"
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: X not in Y?

File: terraform/aio/templates/userdata.cfg.tpl [hunk 7]
```diff
@@ -2,3 +2,11 @@
 # Don't automatically mount ephemeral disk
 mounts:
   - [/dev/vdb, null]
+# WORKAROUND: internal DNS missing from SMS lab.
+runcmd:
+  - 'echo "10.0.0.34 pelican pelican.service.compute.sms-lab.cloud" >> /etc/hosts'
+  - 'echo "10.205.3.187 pulp-server pulp-server.internal.sms-cloud" >> /etc/hosts'
+# Configure SSH keys here, to avoid creating an ephemeral keypair.
+# This means only the instance needs to be cleaned up if the destroy fails.
+ssh_authorized_keys:
```
Comment (markgoddard) [inline terraform/aio/templates/userdata.cfg.tpl:11]: IIRC this didn't work for me in the Ubuntu job, unsure which user got the key.

File: etc/kayobe/ansible/templates/stackhpc-secrets.yml.j2 (hunk 3 above)
Comment (jovial) [reply etc/kayobe/ansible/templates/stackhpc-secrets.yml.j2]: Could work. Are you imagining to just use that for the AIO? Or would we write it some location in kayobe config? Something like `{{ kayobe_config_env_path }}/secrets/libvirt_sasl_password`?

File: etc/kayobe/compute.yml (hunk 4 above)
Comment (jovial) [reply etc/kayobe/compute.yml:182]: Think so, will check when these keys landed upstream.

File: etc/kayobe/compute.yml (hunk 5 above)
Comment (jovial) [reply etc/kayobe/compute.yml]: Can we keep it in the base config? It isn't used unless you set kolla_enable_nova_libvirt_container to false.

File: etc/kayobe/kolla.yml (hunk 6 above)
Comment (jovial) [reply etc/kayobe/kolla.yml]: Much nicer, thanks.

File: terraform/aio/templates/userdata.cfg.tpl (hunk 7 above)
Comment (jovial) [reply terraform/aio/templates/userdata.cfg.tpl:11]: Cheers, I didn't check this as I only saw the ubuntu PR late and thought I'd adopt the method from there.

Comment (jovial) [reply terraform/aio/templates/userdata.cfg.tpl:11]: Fixed with https://github.com/stackhpc/stackhpc-kayobe-config/pull/155/commits/ff692fa94a46d675a7ffc114d9c464c914192db7

File: .github/workflows/stackhpc-pull-request.yml [hunk 8]
```diff
@@ -57,4 +57,34 @@ jobs:
...
+    name: aio (Rocky OVS)
+    needs:
+      - build-kayobe-image
+    uses: ./.github/workflows/stackhpc-all-in-one.yml
+    with:
+      kayobe_image: ${{ needs.build-kayobe-image.outputs.kayobe_image }}
+      os_distribution: rocky
+      neutron_plugin: ovs
+      # FIXME: This image is ancient, but is the only Rocky image uploaded?
+      vm_image: Rocky-8-GenericCloud-8.5-20211114.2.x86_64
```
Comment (jovial) [inline .github/workflows/stackhpc-pull-request.yml]:
```suggestion
      vm_image: Rocky8
```

File: etc/kayobe/compute.yml (hunk 4 above)
Comment (jovial) [reply etc/kayobe/compute.yml:182]: Trying to fix upstream here: https://review.opendev.org/c/openstack/kayobe/+/856454. Will also need syncing to kayobe-config.

File: .github/workflows/stackhpc-all-in-one.yml [hunk 9]
```diff
@@ -56,6 +64,12 @@ jobs:
       KAYOBE_VAULT_PASSWORD: ${{ secrets.KAYOBE_VAULT_PASSWORD }}
       KAYOBE_IMAGE: ${{ inputs.kayobe_image }}
     steps:
+      - name: Set bootstrap user output
+        id: bootstrap_user
+        run: |
+          user=${{ github.event.inputs.vm_user }}
```
Comment (jovial) [inline .github/workflows/stackhpc-all-in-one.yml]:
```suggestion
          user=${{ inputs.vm_user }}
```

File: .github/workflows/stackhpc-all-in-one.yml [hunk 10]
```diff
@@ -151,10 +163,13 @@ jobs:
         run: |
           cat << EOF > etc/kayobe/environments/$KAYOBE_ENVIRONMENT/zz-aio-scenario.yml
           ---
+          os_distribution: ${{ env.OS_DISTRIBUTION }}
           kolla_enable_ovn: ${{ env.ENABLE_OVN }}
+          bootstrap_user: ${{ steps.name.ouputs.value }}
```
Comment (jovial) [inline .github/workflows/stackhpc-all-in-one.yml]:
```suggestion
          bootstrap_user: ${{ steps.bootstrap_user.ouputs.value }}
```

File: etc/kayobe/stackhpc.yml [hunk 11]
```diff
@@ -103,6 +103,18 @@ stackhpc_repo_rabbitmq_server_version: "{{ stackhpc_repo_distribution }}"
 stackhpc_repo_treasuredata_4_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/treasuredata/4/redhat/8/x86_64/{{ stackhpc_repo_treasuredata_4_version }}"
 stackhpc_repo_treasuredata_4_version: "{{ stackhpc_repo_distribution }}"
 
+# Rocky 8 BaseOS
+stackhpc_repo_rocky_baseos_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8/BaseOS/x86_64/os/{{ stackhpc_repo_rocky_baseos_version }}"
```
Comment (jovial) [inline etc/kayobe/stackhpc.yml]:
```suggestion
stackhpc_repo_rocky_baseos_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8.6/BaseOS/x86_64/os/{{ stackhpc_repo_rocky_baseos_version }}"
```

File: etc/kayobe/stackhpc.yml [hunk 12]
```diff
@@ -103,6 +103,18 @@ stackhpc_repo_rabbitmq_server_version: "{{ stackhpc_repo_distribution }}"
...
+stackhpc_repo_rocky_baseos_version: "{{ stackhpc_repo_distribution }}"
+
+# Rocky 8 AppStream
+stackhpc_repo_rocky_appstream_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8/AppStream/x86_64/os/{{ stackhpc_repo_rocky_appstream_version }}"
```
Comment (jovial) [inline etc/kayobe/stackhpc.yml]:
```suggestion
stackhpc_repo_rocky_appstream_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8.6/AppStream/x86_64/os/{{ stackhpc_repo_rocky_appstream_version }}"
```

File: etc/kayobe/stackhpc.yml [hunk 13]
```diff
@@ -103,6 +103,18 @@ stackhpc_repo_rabbitmq_server_version: "{{ stackhpc_repo_distribution }}"
...
+stackhpc_repo_rocky_appstream_version: "{{ stackhpc_repo_distribution }}"
+
+# Rocky 8 extras
+stackhpc_repo_rocky_extras_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8/extras/x86_64/os/{{ stackhpc_repo_rocky_extras_version }}"
```
Comment (jovial) [inline etc/kayobe/stackhpc.yml]:
```suggestion
stackhpc_repo_rocky_extras_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/rocky/8.6/extras/x86_64/os/{{ stackhpc_repo_rocky_extras_version }}"
```

**Comment by jovial** [comment]
Closing in favour of https://github.com/stackhpc/stackhpc-kayobe-config/pull/235.

**Pull request closed without merging**